### DISCUSSION TITLE
QFieldSync object changes in the Android app cannot be synchronized

### DISCUSSION THREAD

**Discussion by Psens666** [opengisch/QField#5007] [Bugs]
Hi,

I have created a project in QGIS and would like to synchronize this with QFielSync in the cloud, unfortunately the whole thing is not working properly. If I have selected the following setting, I do not see any objects on the Objects layer in the app.

![Screenshot 2024-02-05 074556](https://github.com/opengisch/QField/assets/37949999/1fe7eb66-7b8e-49ad-ad6e-21f37aa9eeb9)

![image](https://github.com/opengisch/QField/assets/37949999/ea356bdc-d870-4c72-9faf-c7749dbafcc4)

if I choose the settings differently, see below I see objects, but I can edit them locally on the cell phone, but I cannot synchronize the changes, I am not shown the possibility in the app that changes are available for synchronization .

![image](https://github.com/opengisch/QField/assets/37949999/c7a0622c-b95b-4651-b7bf-31af66bd538a)

![image](https://github.com/opengisch/QField/assets/37949999/25ef9016-943c-418b-a7eb-c1d161551b47)

Can anyone help me here?

Thank you for the answers.

Greetings Dominic

**Comment by suricactus** [comment]
@Psens666 It is not clear if you are trying to use QFieldCloud? If yes, can you please try https://github.com/opengisch/QField/pull/5118 and share your experience?

**Comment by 8Daniel3** [comment]
Hello, i have exactly the same problem an Psens666 (also trying to edit field borders in bavaria ;-) )

Normally, I create a border (polygon) in QGIS, sync in with QFieldCloud, edit the polygon outside in the field with my smartphone, resync it with QFieldCloud and then i have the exact borders in QGIS for my fields.

This worked well for a while, but since my last try i can edit the borders with my smartphone but if i want to push the changes to the cloud only "syncronize" is available. Then, if i syncronize, my changes are gone and i have the unedited version from the cloud on my smartphone.

When the layer is set to "offline editing" i can't see the polygons (object list is empty), when it's set to "directly access data source" i can see the polygons and edit them but can't "save" them to the cloud as described above.

Now i tested to create a new project with a new shape layer, inserted the coordinates from a field with the extension "NumericalDigitize" as usual, uploaded it to QFieldCloud (new project / "offline editing" is selected as default), syncronized it to the smartphone, edited the polygon and i could push the changes to the cloud to sync it back to QGIS. 

So, with a new project, everything works as it should! It has to be just a little thing to change that the polygons are visible to edit, but i don't know what it is.

**Reply by Psens666** [reply]
Hallo Daniel,

ich versuch es mal auf deutsch, bayrisch oder in meinem fall frängisch wird wohl schlecht zu lesen sein :D

Ich habe festgestellt das man immer den OpenStreetMap Layer im Projekt haben muss dann funktioniert alles richtig, warum auch immer :D. 

Zumindest war das die Lösung bei mir. 

![Screenshot 2024-04-16 135145](https://github.com/opengisch/QField/assets/37949999/8a47c789-7dd0-4623-b7da-58d44bf527ae)

**Reply by 8Daniel3** [reply]
Mit frängisch komm ich gut klar. Grüße aus den Hassbergen ;-)

Habe OSM als Layer hinzugefügt, das Problem ist aber bei mir immer noch vorhanden.
"directly access data source" -> Polygone da, aber nicht änderbar
"offline editing" -> keine Polygone da
